kolibri/plugins/facility/assets/src/views/users/sidePanels/AssignCoachesSidePanel.vue

```diff
@@ -1,35 +1,347 @@
+      :submitDisabled="isLoading"
+      :cancelDisabled="isLoading"
+      @cancel="handleDismissConfirmation"
+      @submit="handleUndoAssignments"
+    >
+      <KCircularLoader v-if="isLoading" />
+      <span
+        v-else
+        class="adjust-line-height"
+      >
+        {{
+          undoAssignCoachMessage$({
+            numUsers: eligibleUsersCount,
+            numClasses: selectedClasses.length,
+          })
+        }}
+      </span>
+    </KModal>
+    <SidePanelModal
+      v-else
+      alignment="right"
+      sidePanelWidth="700px"
+      :addBottomBorder="false"
+      @closePanel="closeSidePanel(selectedClasses.length > 0)"
+    >
+      <template #header>
+        <h1>{{ assignUsersHeading$({ num: selectedUsersCount }) }}</h1>
+      </template>
+
+      <div class="assign-coaches-content">
+        <KCircularLoader v-if="isLoading" />
+        <div v-else>
+          <div
+            v-if="showErrorWarning"
+            class="assign-warning-label"
+            :style="{ color: $themeTokens.error }"
+          >
+            <span>{{ defaultErrorMessage$() }}</span>
+          </div>
+
+          <div
+            class="top-info-box"
+            :style="{ backgroundColor: $themePalette.grey.v_100 }"
+          >
+            <template v-if="ineligibleUsersCount > 0">
+              <div class="info-flex">
+                <KIcon
+                  icon="infoOutline"
+                  class="info-icon"
+                />
+                <div class="info-lines">
+                  <div class="info-line">
+                    {{ numUsersNotEligible$({ num: ineligibleUsersCount }) }}
+                  </div>
+                  <div class="info-line">{{ usersInClassNotAffected$() }}</div>
+                </div>
+              </div>
+            </template>
+            <template v-else>
+              <div class="info-lines">
+                <div class="info-line">{{ usersInClassNotAffected$() }}</div>
+              </div>
+            </template>
+          </div>
+
+          <h2>{{ selectClassesLabel$() }}</h2>
+          <SelectableList
+            v-model="selectedClasses"
+            :options="formattedClasses"
+            aria-labelledby="classes-heading"
+            :selectAllLabel="assignToAllClasses$()"
+            :searchLabel="searchForAClass$()"
+          />
+        </div>
+      </div>
+      <template #bottomNavigation>
+        <div class="bottom-nav-container">
+          <KButtonGroup>
+            <KButton
+              :text="coreString('cancelAction')"
+              :disabled="isLoading"
+              @click="closeSidePanel(selectedClasses.length > 0)"
+            />
+            <KButton
+              primary
+              :text="assignAction$()"
+              :disabled="!hasSelectedClasses || isLoading"
+              @click="handleAssign"
+            />
+          </KButtonGroup>
+        </div>
+      </template>
+
+      <KModal
+        v-if="showCloseConfirmationModal"
```

Comment:
We have this [CloseConfirmationGuard](https://github.com/lianaharris360/kolibri/blob/ccda0041b98f14fb0a7dd2f513b21252226ad48e/kolibri/plugins/facility/assets/src/views/users/common/CloseConfirmationGuard.vue#L28) component to abstract the confirmation modal logic.

[Here](https://github.com/lianaharris360/kolibri/blob/ccda0041b98f14fb0a7dd2f513b21252226ad48e/kolibri/plugins/facility/assets/src/views/users/sidePanels/UserCreate/index.vue#L109) is an example of how we can use it. (Just noticed the beforeRouteLeave link that needs to be done is not well docummented in the component 😅).